unity/EditorPlugin/AssetPostprocessors/CsprojAssetPostprocessor.cs

```diff
@@ -509,7 +509,8 @@ private static bool SetLangVersion(XElement projectElement, XNamespace xmlns)
         // Note that if existing is "default", we'll override it
         var currentIsParsed = VersionExtensions.TryParse(existing, out var currentLanguageLevel);
         var expectedIsParsed = VersionExtensions.TryParse(expected, out var expectedLanguageLevel);
-        if (currentIsParsed && expectedIsParsed && currentLanguageLevel < expectedLanguageLevel)
+        if (currentIsParsed && expectedIsParsed && currentLanguageLevel < expectedLanguageLevel
+            || !currentIsParsed)
```

Comment:
Presumably this fails because `LangVersion` isn't even set in the file (I think later versions of Unity started writing this themselves), so the incoming `existing` value will be `string.Empty`. We don't handle empty strings very well, here or in `SetOrUpdateProperty`.

I think we should be more explicit about our intentions when we get an empty string - if it's unset, always set it to `expected`:

```
if (string.IsNullOrEmpty(existing))
  return expected;
```

And then just leave this `if` statement as it was.

Finally, we should also change `SetOrUpdateProperty` to not write or update anything if the returned value is `string.Empty` -  this would stop us getting empty elements. Perhaps with some logging.

Reply:
Adding Property with empty value was never a case, but it may still be valid.

Reply:
Yeah, that comment was wrong before any of these changes 😊 What do you think we should do for `"default"`? Just leave it, like we do with `"latest"`, or replace it with `expected`?